### PR TITLE
Fix AML when no users assigned a role, and upgrade Porter and TF mixin

### LINKED ISSUE #2549: AML Workspace service URL is incorrect
URL created by AML Workspace service contains Unicode character that fails in navigation

**Steps to reproduce**

1. Create AML Workspace service
2.  Connection URI contains Unicode "\\u0026" instead of "&" sign, example:       "connection_uri": "https://ml.azure.com/?wsid=/subscriptions/xxxx/resourceGroups/rg-scctre-ws-579a/providers/Microsoft.MachineLearningServices/workspaces/ml-scctre-ws-579a-svc-28b0\\u0026tid=xxxxxx"

3. Navigation fails

### PR DESCRIPTION
**Pull request by marrobi**
# Resolves #2549 and #2626

## What is being addressed
- Upgrade Porter to 0.38.13 in dev container and resource processor
- Manually install latest Porter Terraform mixin in dev container
- Do not assing AML roles if no users
- [x] Change log

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -54,13 +54,14 @@ RUN apt-get update && apt-get install -y python3 python3-venv libaugeas0 --no-in
...
 ARG PORTER_MIRROR=https://cdn.porter.sh
-ARG PORTER_PERMALINK=v0.38.12
+ARG PORTER_PERMALINK=v0.38.13
 ARG PORTER_PKG_PERMALINK=latest
 ARG PORTER_HOME=/home/$USERNAME/.porter/
 COPY .devcontainer/scripts/porter.sh /tmp/
 RUN export PORTER_MIRROR=${PORTER_MIRROR} \
     PORTER_PERMALINK=${PORTER_PERMALINK} \
     PORTER_PKG_PERMALINK=${PORTER_PKG_PERMALINK} \
+    PORTER_TERRAFORM_MIXIN_PKG_PERMALINK="v1.0.0-rc.1" \
```
Comment (martinpeck) [inline .devcontainer/Dockerfile:64]: should this be defined up above too (next to the place we define `latest`)?

Comment (martinpeck) [reply .devcontainer/Dockerfile:64]: this is a "nit" and ignorable

File: templates/workspace_services/azureml/terraform/roles.tf [hunk 2]
```diff
@@ -20,7 +20,7 @@ data "azurerm_role_definition" "azure_ml_data_scientist" {
 }
 
 resource "azurerm_role_assignment" "app_role_members_aml_data_scientist" {
-  for_each           = toset(split("\n", data.external.app_role_members.result.principals))
+  for_each           = (data.external.app_role_members.result.principals == "") ? [] : toset(split("\n", data.external.app_role_members.result.principals))
```
Comment (martinpeck) [inline templates/workspace_services/azureml/terraform/roles.tf:23]: I haven't reviewed this (not sure what it's doing)

**Review by martinpeck — APPROVED** [review]
made some comments

File: templates/workspace_services/azureml/terraform/roles.tf (hunk 2 above)
Comment (marrobi) [reply templates/workspace_services/azureml/terraform/roles.tf:23]: I've tested it and works, if the principals string is empty gives an empty array so no role assignments are made. Previously it was trying to use an empty string as an object id. If there are principals, then works as before.

**Comment by marrobi** [comment]
/test-force-approve passed https://github.com/microsoft/AzureTRE/actions/runs/3108570702

**Pull request merged**